### PR TITLE
Added README for ONNX Frontend

### PR DESCRIPTION
**Pull request by mbencer**
Ticket: 82498

### CODE REVIEW

File: src/frontends/onnx/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# ONNX FE
+
+The main responsibility of ONNX Frontend is importing ONNX models and conversion them to the `ov::Model` representation.
```
Comment (t-jankowski) [inline src/frontends/onnx/README.md:3]: Either 'importing and converting' or 'import of and conversion of'.
Rather use 'these' while referring to non-living matter.
My choice would be 'import of ONNX models and conversion of these into the ...'

File: src/frontends/onnx/README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# ONNX FE
+
+The main responsibility of ONNX Frontend is importing ONNX models and conversion them to the `ov::Model` representation. 
+To capabilities of ONNX Frontend belong also:
```
Comment (t-jankowski) [inline src/frontends/onnx/README.md:4]:
```suggestion
Other capabilities of ONNX Frontend:
```

File: src/frontends/onnx/README.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+* modification of tensors properties (like data type and shapes)
+* changing topology of the models (like cutting subgraphs, inserting additional inputs and outputs)
+* user-friendly searching the models (via tensors and operators names)
+
+The component is written in `C++`, but `Python` bindings are also available.
```
Comment (t-jankowski) [inline src/frontends/onnx/README.md:9]:
```suggestion
The component is written in `C++`, `Python` bindings are also available.
```

**Review by t-jankowski — CHANGES_REQUESTED** [review]

File: src/frontends/onnx/README.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+* Mateusz Tabaka (mateusztabaka)
+* Mateusz Bencer (mbencer)
+
+from `openvino-onnx-frontend-maintainers` maintainers group.
+
+
+## Components
+
+ONNX Frontend implements an interface common for all frontends defined in [Frontends API](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/common/include/openvino/frontend).
+The exported symbols are decorated with `ONNX_FRONTEND_API` flag. The key class responsible for importing is `FrontEnd` defined in (ONNX Frontend class)[https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/include/openvino/frontend/onnx/frontend.hpp].
```
Comment (tomdol) [inline src/frontends/onnx/README.md:26]: I think this sentence should be "prefixed" with an explanation that ONNX FE is a (shared) library.
```suggestion
The exported symbols are decorated with `ONNX_FRONTEND_API` macro. The key class responsible for importing is `FrontEnd` defined in (ONNX Frontend class)[https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/include/openvino/frontend/onnx/frontend.hpp].
```

Comment (tomdol) [reply src/frontends/onnx/README.md:26]: BTW I'm not sure I understand "The key class responsible for importing is `FrontEnd`" - can you rephrase it?

File: src/frontends/onnx/README.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+For backward compatibility reasons, the ONNX importer API (more lower-level abstraction approach) is still maintained. It can be found in (ONNX Importer)[https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/include/onnx_import/onnx.hpp]. The symbols of ONNX Importer API are decorated with `ONNX_IMPORTER_API` flag.
+
+The crucial place in ONNX Frontend is (ops implementation folder)[https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/frontend/src/op], where the operators are implemented. Each built-in operator has to be registered in ops bridge - (ops bridge)[https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/ops_bridge.cpp]. Expect that, ONNX frontend gives capabilities to register a custom op by an user. It can be achieved via `ConversionExtension` available in (Conversion)[https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/include/openvino/frontend/onnx/extension/conversion.hpp].
```
Comment (tomdol) [inline src/frontends/onnx/README.md:29]:
```suggestion
The crucial place in ONNX Frontend is the directory where the operators are implemented:[https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/frontend/src/op]. Each operator handler has to be registered in ops bridge - (ops bridge)[https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/ops_bridge.cpp]. Expect that ONNX frontend has capabilities to register a custom op by a user. It can be achieved via `ConversionExtension` available in (Conversion)[https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/include/openvino/frontend/onnx/extension/conversion.hpp].
```

File: src/frontends/onnx/README.md (hunk 2 above)
Comment (mbencer) [reply src/frontends/onnx/README.md:4]: done

File: src/frontends/onnx/README.md (hunk 3 above)
Comment (mbencer) [reply src/frontends/onnx/README.md:9]: done

File: src/frontends/onnx/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mbencer) [reply src/frontends/onnx/README.md:3]: applied @t-jankowski suggestions - "is import of ONNX models and conversion of these into the..". Does it work for you?

File: src/frontends/onnx/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mbencer) [reply src/frontends/onnx/README.md:4]: done

File: src/frontends/onnx/README.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+## Tutorials
+TBD
+
+
+## See also
+
+ * [ONNX standard repository](https://github.com/onnx/onnx)
+ * [ONNX operators list](https://github.com/onnx/onnx/blob/main/docs/Operators.md)
+ * [ONNX Runtime OpenVINO Provider](https://github.com/microsoft/onnxruntime/tree/cccbe90764616749f06633fadaf90aba2eb360c9/onnxruntime/core/providers/openvino)
```
[the remark this answers is not in the document]
Comment (mbencer) [reply src/frontends/onnx/README.md:54]: good catch, changed to master

File: src/frontends/onnx/README.md (hunk 4 above)
Comment (mbencer) [reply src/frontends/onnx/README.md:26]: I wanted to say like "the heart of the ONNX FE ecosystem if Frontend class", but now I'm thinking that it is rather useless info for user/contributor so I've removed this sentence.

File: src/frontends/onnx/README.md (hunk 5 above)
Comment (mbencer) [reply src/frontends/onnx/README.md:29]: done

**Review by t-jankowski — APPROVED** [review]

File: src/frontends/onnx/README.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+* Mateusz Bencer (mbencer)
+
+from `openvino-onnx-frontend-maintainers` maintainers group.
+
+
+## Components
+
+ONNX Frontend implements an interface common for all frontends defined in [Frontends API](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/common/include/openvino/frontend).
+The exported symbols are decorated with `ONNX_FRONTEND_API` macro.
+For backward compatibility reasons, the ONNX importer API (more lower-level abstraction approach) is still maintained. It can be found in (ONNX Importer)[https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/include/onnx_import/onnx.hpp]. The symbols of ONNX Importer API are decorated with `ONNX_IMPORTER_API` macro.
```
Comment (ilyachur) [inline src/frontends/onnx/README.md:27]: It is low level details, I think it can be described on dedicated page inside the docs folder.
The idea is to don't overload user by information about the product, here we can have links to dedicated pages and he will read in case of needed.

File: src/frontends/onnx/README.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+The crucial place in ONNX Frontend is the directory where the operators are implemented:[https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/frontend/src/op]. Each operator handler has to be registered in ops bridge - (ops bridge)[https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/ops_bridge.cpp]. Expect that ONNX frontend has capabilities to register a custom op by a user. It can be achieved via `ConversionExtension` available in (Conversion)[https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/include/openvino/frontend/onnx/extension/conversion.hpp].
```
Comment (ilyachur) [inline src/frontends/onnx/README.md:29]: These links don't render well.

File: src/frontends/onnx/README.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+API of ONNX Frontend can be called directly, but it is also used internally by (Model Optimizer)[https://github.com/openvinotoolkit/openvino/tree/master/tools/mo] during ONNX to IR (Intermediate Representation) conversion. What's more capabilities of ONNX Frontend are used by ONNX Runtime via OpenVINO Execution Provider (more information can be found in (docs)[https://onnxruntime.ai/docs/build/eps.html#openvino]).
```
Comment (ilyachur) [inline src/frontends/onnx/README.md:31]: Please use relative paths. It will allow to have a branch independent documentation.

File: src/frontends/onnx/README.md [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+ONNX Frontend is tested in few places:
+- [C++ gtest-based tests](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/tests)
+- [Python frontend tests](https://github.com/openvinotoolkit/openvino/tree/master/src/bindings/python/tests/test_frontend)
+- [Python operators tests](https://github.com/openvinotoolkit/openvino/tree/master/src/bindings/python/tests/test_onnx)
+- [Python compliance with ONNX standard tests](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/tests/test_onnx/test_backend.py)
+- [Python OpenModelZoo tests](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/tests/test_onnx/test_zoo_models.py)
```
Comment (ilyachur) [inline src/frontends/onnx/README.md:38]: I think for tests we can have a dedicated page

File: src/frontends/onnx/README.md [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+
+
+## Architecture
+The overview of components responsible for importing scenario is shown on the diagram below:
+![ONNX overview diagram](docs/img/onnx_fe_overview.png)
+
+
+## Tutorials
+TBD
```
Comment (ilyachur) [inline src/frontends/onnx/README.md:47]:
```suggestion
* How to add new operation
* How to test
* How to debug
* ...
```

**Review by ilyachur — COMMENTED** [review]
Please update link on the page: https://github.com/openvinotoolkit/openvino/blob/master/src/README.md

File: src/frontends/onnx/README.md [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+- [C++ gtest-based tests](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/tests)
+- [Python frontend tests](https://github.com/openvinotoolkit/openvino/tree/master/src/bindings/python/tests/test_frontend)
+- [Python operators tests](https://github.com/openvinotoolkit/openvino/tree/master/src/bindings/python/tests/test_onnx)
+- [Python compliance with ONNX standard tests](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/tests/test_onnx/test_backend.py)
+- [Python OpenModelZoo tests](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/tests/test_onnx/test_zoo_models.py)
+
+
+## Architecture
+The overview of components responsible for importing scenario is shown on the diagram below:
+![ONNX overview diagram](docs/img/onnx_fe_overview.png)
```
[the remark this answers is not in the document]
Comment (mbencer) [reply src/frontends/onnx/README.md:43]: done

File: src/frontends/onnx/README.md (hunk 7 above)
Comment (mbencer) [reply src/frontends/onnx/README.md:27]: I see, I've removed some details (like exporting flags), but still keep some very basic info about op registering (to consider when tutorial - how to add a new op will be ready)

File: src/frontends/onnx/README.md (hunk 8 above)
Comment (mbencer) [reply src/frontends/onnx/README.md:29]: good catch, fixed

File: src/frontends/onnx/README.md (hunk 10 above)
Comment (mbencer) [reply src/frontends/onnx/README.md:38]: move to the tests tutorial

File: src/frontends/onnx/README.md (hunk 11 above)
Comment (mbencer) [reply src/frontends/onnx/README.md:47]: done

**Comment by mbencer** [comment]
@ilyachur Thank you for many useful tips ;)

File: src/frontends/onnx/README.md [hunk 13]
```diff
@@ -0,0 +1,54 @@
+# ONNX FE
+
+The main responsibility of ONNX Frontend is import of ONNX models and conversion of these into the `ov::Model` representation. 
+Other capabilities of the ONNX Frontend:
+* modification of tensors properties (like data type and shapes)
+* changing topology of the models (like cutting subgraphs, inserting additional inputs and outputs)
+* user-friendly searching the models (via tensors and operators names)
+
+The component is written in `C++`, `Python` bindings are also available.
+Each change to ONNX FE component requires `clang-format` code style and `NCC` naming style checks.
```
[the remark this answers is not in the document]
Comment (mbencer) [reply src/frontends/onnx/README.md:10]: done

**Comment by mbencer** [comment]
> Please update link on the page: https://github.com/openvinotoolkit/openvino/blob/master/src/README.md

Do you mean openvino/src/frontends/README.md? It yes, updated

File: src/frontends/onnx/README.md (hunk 7 above)
Comment (ilyachur) [reply src/frontends/onnx/README.md:27]: Can we do this page now? :)
I added documentation for IR frontend: https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/ir/README.md

Maybe you can also take a look to this page. Here we don't provide any details on  the main page, for details dedicated pages are created

File: src/frontends/onnx/README.md [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+![ONNX overview diagram](docs/img/onnx_fe_overview.png)
+
+
+## Tutorials
+TBD
+
+
+## See also
+
+ * [ONNX standard repository](https://github.com/onnx/onnx)
```
[the remark this answers is not in the document]
Comment (mbencer) [reply src/frontends/onnx/README.md:52]: done

File: src/frontends/onnx/README.md [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+
+In case of any questions, review and merge requests, please contact [openvino-onnx-frontend-maintainers](https://github.com/orgs/openvinotoolkit/teams/openvino-onnx-frontend-maintainers) group.
+
+
+## Components
+
+ONNX Frontend implements an interface common for all frontends defined in [Frontends API](./src/frontends/common/include/openvino/frontend).
+For backward compatibility reasons, the ONNX importer API (more lower-level abstraction approach) is still maintained. It can be found in [ONNX Importer](.src/frontends/onnx/frontend/include/onnx_import/onnx.hpp).
+
+The crucial place in ONNX Frontend is [the directory](.src/frontends/onnx/frontend/src/op) where the operators are implemented. Each operator handler has to be registered in [ops bridge](.src/frontends/onnx/frontend/src/ops_bridge.cpp). More details about it you can find in [How to add an new operation](./docs/how_to_add_op.md). Expect that ONNX frontend has capabilities to register a custom op by a user. It can be achieved via [ConversionExtension](.include/openvino/frontend/onnx/extension/conversion.hpp).
```
Comment (ilyachur) [inline src/frontends/onnx/README.md:23]:
```suggestion
The crucial place in ONNX Frontend is [the directory](./src/frontends/onnx/frontend/src/op) where the operators are implemented. Each operator handler has to be registered in [ops bridge](./src/frontends/onnx/frontend/src/ops_bridge.cpp). More details about it you can find in [How to add an new operation](./docs/how_to_add_op.md). Expect that ONNX frontend has capabilities to register a custom op by a user. It can be achieved via [ConversionExtension](./include/openvino/frontend/onnx/extension/conversion.hpp).
```

File: src/frontends/onnx/README.md [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+
+API of ONNX Frontend can be called directly, but it is also used internally by [Model Optimizer](https://github.com/openvinotoolkit/openvino/tree/master/tools/mo) during ONNX to IR (Intermediate Representation) conversion. What's more capabilities of ONNX Frontend are used by [ONNX Runtime via OpenVINO Execution Provider](https://onnxruntime.ai/docs/build/eps.html#openvino).
```
Comment (ilyachur) [inline src/frontends/onnx/README.md:25]: Please use relative links

```suggestion
API of ONNX Frontend can be called directly, but it is also used internally by [Model Optimizer](../../../tools/mo) during ONNX to IR (Intermediate Representation) conversion. What's more capabilities of ONNX Frontend are used by [ONNX Runtime via OpenVINO Execution Provider](https://onnxruntime.ai/docs/build/eps.html#openvino).
```

File: src/frontends/onnx/README.md [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+        others["..."]
+        set_pt_shape["set_partial_shape"]
+        add_output["add_output"]
+        extract_sub["extract_subgraph"]
+    end
+
+    subgraph Frontend
+        fe_load["load_impl"]
+        fe_convert["convert"]
+        fe_others["..."]
```
Comment (ilyachur) [inline src/frontends/onnx/README.md:50]: Why do we need all these methods? 
Maybe we can show only used methods like here: https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/ir/docs/architecture.md

Or maybe we can create a separate document for architecture and say more words about different classes and their usage?

File: src/frontends/onnx/docs/tests.md [hunk 18]
```diff
@@ -0,0 +1,12 @@
+# ONNX Frontend tests
+
+## <a name="places">Places where ONNX FE is tested
+- [C++ gtest-based tests](tests)
```
Comment (ilyachur) [inline src/frontends/onnx/docs/tests.md:4]:
```suggestion
- [C++ gtest-based tests](../tests)
```

File: src/frontends/onnx/docs/how_to_add_op.md [hunk 19]
```diff
@@ -0,0 +1,2 @@
+# How to add an new operation
+TBD
```
Comment (ilyachur) [inline src/frontends/onnx/docs/how_to_add_op.md:2]: Do we have a ticket to cover all TBDs in future?

File: src/frontends/onnx/README.md (hunk 7 above)
Comment (mbencer) [reply src/frontends/onnx/README.md]: done, dedicated page (TBD - 93001) created, details in the main page removed

File: src/frontends/onnx/README.md (hunk 17 above)
Comment (mbencer) [reply src/frontends/onnx/README.md]: I wanted to show editing capabilities of ONNX Frontend, which are not exposed enough in my opinion. But agree, that class diagram in a separate document is better here - ticket 93001

File: src/frontends/onnx/docs/how_to_add_op.md (hunk 19 above)
Comment (mbencer) [reply src/frontends/onnx/docs/how_to_add_op.md]: yes, just as summary:
93001 - how to add a new op
93002 - ONNX FE tests
94113 - class diagram

btw, a great idea of component doc ;) I have a plan to fill sub pages content after FF

**Comment by ilyachur** [comment]
@tsavina Could you please take a look?

File: src/frontends/onnx/README.md [hunk 20]
```diff
@@ -0,0 +1,66 @@
...
+* changing topology of the models (like cutting subgraphs, inserting additional inputs and outputs)
+* user-friendly searching the models (via tensors and operators names)
+
+The component is written in `C++`, `Python` bindings are also available.
+If you want to contribute, please follow [the common coding style rules](../../../docs/dev/coding_style.md).
+
+
+## Key contacts
+
+In case of any questions, review and merge requests, please contact [openvino-onnx-frontend-maintainers](https://github.com/orgs/openvinotoolkit/teams/openvino-onnx-frontend-maintainers) group.
```
Comment (tsavina) [inline src/frontends/onnx/README.md:15]:
```suggestion
In case of any questions, review and merge requests, contact the [openvino-onnx-frontend-maintainers](https://github.com/orgs/openvinotoolkit/teams/openvino-onnx-frontend-maintainers) group.
```

File: src/frontends/onnx/README.md [hunk 21]
```diff
@@ -0,0 +1,66 @@
...
+
+
+## Components
+
+ONNX Frontend implements an interface common for all frontends defined in [Frontends API](./src/frontends/common/include/openvino/frontend).
```
Comment (tsavina) [inline src/frontends/onnx/README.md:20]:
```suggestion
ONNX Frontend implements an interface common to all frontends defined in the [Frontends API](./src/frontends/common/include/openvino/frontend).
```

File: src/frontends/onnx/README.md [hunk 22]
```diff
@@ -0,0 +1,66 @@
...
+For backward compatibility reasons, the ONNX importer API (more lower-level abstraction approach) is still maintained. It can be found in [ONNX Importer](.src/frontends/onnx/frontend/include/onnx_import/onnx.hpp).
```
Comment (tsavina) [inline src/frontends/onnx/README.md:21]:
```suggestion
For backward compatibility reasons, the ONNX importer API (more lower-level abstraction approach) is still maintained. You can find it in the [ONNX Importer](.src/frontends/onnx/frontend/include/onnx_import/onnx.hpp).
```

Comment (tsavina) [inline src/frontends/onnx/README.md:21]: Links don't work

File: src/frontends/onnx/README.md [hunk 23]
```diff
@@ -0,0 +1,66 @@
...
+
+The information about adding new operations to ONNX Frontend (both contribution and extension scenario) can be found in [How to add an new operation](./docs/how_to_add_op.md).
```
Comment (tsavina) [inline src/frontends/onnx/README.md:23]:
```suggestion
Read [How to add a new operation](./docs/how_to_add_op.md) page for more information on adding new operations to the ONNX Frontend and extending the existing ones. 
```

File: src/frontends/onnx/README.md [hunk 24]
```diff
@@ -0,0 +1,66 @@
...
+
+The API of ONNX Frontend can be called directly, but it is also used internally by [Model Optimizer](../../../tools/mo) during ONNX to IR (Intermediate Representation) conversion. What's more capabilities of ONNX Frontend are used by [ONNX Runtime via OpenVINO Execution Provider](https://onnxruntime.ai/docs/build/eps.html#openvino).
```
Comment (tsavina) [inline src/frontends/onnx/README.md:25]:
```suggestion
The API of ONNX Frontend can be called directly. It is also used internally by [Model Optimizer](../../../tools/mo) during the conversion from ONNX to Intermediate Representation (IR). The capabilities of ONNX Frontend are used by the [ONNX Runtime via OpenVINO Execution Provider](https://onnxruntime.ai/docs/build/eps.html#openvino).
```

File: src/frontends/onnx/README.md [hunk 25]
```diff
@@ -0,0 +1,66 @@
...
+
+ONNX Frontend is tested in many places (both `C++` and `Python` tests). More details about testing you can find in [ONNX FE tests](./docs/tests#places)
```
Comment (tsavina) [inline src/frontends/onnx/README.md:27]:
```suggestion
Both `C++` and `Python` tests are implemented for the ONNX Frontend. Read the [ONNX Frontend tests](./docs/tests#places) page for more details. 
```

Comment (tsavina) [inline src/frontends/onnx/README.md:27]: link doesn't work

File: src/frontends/onnx/README.md [hunk 26]
```diff
@@ -0,0 +1,66 @@
...
+
+
+## Architecture
+The overview of components responsible for the basic flow which is a model importing is shown on the diagram below:
```
Comment (tsavina) [inline src/frontends/onnx/README.md:31]:
```suggestion
The diagram below shows an overview of the components responsible for the basic model importing flow:
```

File: src/frontends/onnx/README.md [hunk 27]
```diff
@@ -0,0 +1,66 @@
...
+
+    style onnx fill:#6c9f7f
+    style ov_model fill:#6c9f7f
+
+    onnx-->|protobuf|fe_load
+    fe_load-->InputModel
+    InputModel-->fe_convert
+    fe_convert-->ov_model
+```
+More details about architecture can be found in [ONNX Frontend architecture](./docs/architecture.md).
```
Comment (tsavina) [inline src/frontends/onnx/README.md:54]:
```suggestion
Find more details about the architecture in the [ONNX Frontend architecture](./docs/architecture.md) document.
```

File: src/frontends/onnx/README.md [hunk 28]
```diff
@@ -0,0 +1,66 @@
...
+
+## Tutorials
+* [How to add an new operation](./docs/how_to_add_op.md)
+* [How ONNX FE is tested](./docs/tests.md)
```
Comment (tsavina) [inline src/frontends/onnx/README.md:58]:
```suggestion
* [How to add a new operation](./docs/how_to_add_op.md)
* [How ONNX Frontend is tested](./docs/tests.md)
```

File: src/frontends/onnx/docs/architecture.md [hunk 29]
```diff
@@ -0,0 +1,10 @@
+# ONNX Frontend architecture
+
+On the class diagram below is shown structure and relations between basic classes of ONNX Frontend:
```
Comment (tsavina) [inline src/frontends/onnx/docs/architecture.md:3]:
```suggestion
The class diagram below shows the structure and relations between the basic classes of ONNX Frontend:
```

File: src/frontends/onnx/docs/how_to_add_op.md [hunk 30]
```diff
@@ -0,0 +1,13 @@
+# How to add an new operation
+
+## How to implement an new operation
+
+## How to register an custom operation via Extension mechanism
```
Comment (tsavina) [inline src/frontends/onnx/docs/how_to_add_op.md:5]:
```suggestion
# How to add a new operation

## How to implement a new operation

## How to register a custom operation via Extension mechanism
```

File: src/frontends/onnx/docs/tests.md [hunk 31]
```diff
@@ -0,0 +1,12 @@
+# ONNX Frontend tests
+
+## <a name="places">Places where ONNX FE is tested
```
Comment (tsavina) [inline src/frontends/onnx/docs/tests.md:3]:
```suggestion
## <a name="places">ONNX Frontend testing places
```

File: src/frontends/onnx/README.md (hunk 22 above)
Comment (mbencer) [reply src/frontends/onnx/README.md]: fixed

File: src/frontends/onnx/README.md (hunk 25 above)
Comment (mbencer) [reply src/frontends/onnx/README.md]: fixed

File: src/frontends/onnx/README.md [hunk 32]
```diff
@@ -0,0 +1,66 @@
+# OpenVINO ONNX Frontend
+
+The main responsibility of the ONNX Frontend is to import ONNX models and convert them into the `ov::Model` representation. 
+Other capabilities of the ONNX Frontend:
+* modification of tensors properties (like data type and shapes)
+* changing topology of the models (like cutting subgraphs, inserting additional inputs and outputs)
+* searching the models in a user-friendly way via tensors and operators names
+
+The component is written in `C++`. `Python` bindings are also available.
```
Comment (ilyachur) [inline src/frontends/onnx/README.md:9]: What do you mean: "`Python` bindings are also available."? If I remember right we don't have ONNX Frontend specific objects in python (excepts the extension)

File: src/frontends/onnx/docs/tests.md [hunk 33]
```diff
@@ -0,0 +1,12 @@
+# ONNX Frontend tests
+
+## <a name="places">ONNX Frontend testing places
```
Comment (ilyachur) [inline src/frontends/onnx/docs/tests.md:3]:
```suggestion
## ONNX Frontend testing places
```

File: src/frontends/onnx/README.md [hunk 34]
```diff
@@ -0,0 +1,66 @@
...
+## Components
+
+ONNX Frontend implements an interface common to all frontends defined in the [Frontends API](../common/include/openvino/frontend).
+For backward compatibility reasons, the ONNX importer API (more lower-level abstraction approach) is still maintained. You can find it in the [ONNX Importer](./frontend/include/onnx_import/onnx.hpp).
+
+Read [How to add a new operation](./docs/how_to_add_op.md) page for more information on adding new operations to the ONNX Frontend and extending the existing ones. 
+
+The API of ONNX Frontend can be called directly. It is also used internally by [Model Optimizer](../../../tools/mo) during the conversion from ONNX to Intermediate Representation (IR). The capabilities of ONNX Frontend are used by the [ONNX Runtime via OpenVINO Execution Provider](https://onnxruntime.ai/docs/build/eps.html#openvino).
+
+Both `C++` and `Python` tests are implemented for the ONNX Frontend. Read the [ONNX Frontend tests](./docs/tests.md#places) page for more details.
```
Comment (ilyachur) [inline src/frontends/onnx/README.md:27]:
```suggestion
Both `C++` and `Python` tests are implemented for the ONNX Frontend. Read the [ONNX Frontend tests](./docs/tests.md#onnx-frontend-testing-places) page for more details. 
```
If I remember right it also should work

File: src/frontends/onnx/README.md [hunk 35]
```diff
@@ -0,0 +1,66 @@
...
+
+In case of any questions, review and merge requests, contact the [openvino-onnx-frontend-maintainers](https://github.com/orgs/openvinotoolkit/teams/openvino-onnx-frontend-maintainers) group.
+
+
+## Components
+
+ONNX Frontend implements an interface common to all frontends defined in the [Frontends API](../common/include/openvino/frontend).
+For backward compatibility reasons, the ONNX importer API (more lower-level abstraction approach) is still maintained. You can find it in the [ONNX Importer](./frontend/include/onnx_import/onnx.hpp).
+
+Read [How to add a new operation](./docs/how_to_add_op.md) page for more information on adding new operations to the ONNX Frontend and extending the existing ones.
```
Comment (ilyachur) [inline src/frontends/onnx/README.md:23]: This sentence is not about components, maybe we shouldn't duplicate it here and keep only in the tutorials section.

Comment (mbencer) [reply src/frontends/onnx/README.md:23]: removed

File: src/frontends/onnx/README.md (hunk 34 above)
Comment (mbencer) [reply src/frontends/onnx/README.md:27]: right, it works, done

File: src/frontends/onnx/README.md (hunk 32 above)
Comment (mbencer) [reply src/frontends/onnx/README.md]: I mean here common FE API bindigs here, but you are right, that it can be confused. Removed.

**Review by ilyachur — APPROVED** [review]
We need to extend documentation: tests, how to add new op, architecture
@tsavina please take a look

File: src/frontends/onnx/README.md [hunk 36]
```diff
@@ -0,0 +1,63 @@
...
+
+    style onnx fill:#6c9f7f
+    style ov_model fill:#6c9f7f
+
+    onnx-->|protobuf|fe_load
+    fe_load-->InputModel
+    InputModel-->fe_convert
+    fe_convert-->ov_model
+```
+Find more details about the architecture in the [ONNX Frontend architecture](./docs/architecture.md) document.
```
Comment (tsavina) [inline src/frontends/onnx/README.md:51]:
```suggestion
Find more details in the [ONNX Frontend architecture](./docs/architecture.md) document.
```

**Review by tsavina — APPROVED** [review]
